src/components/victory-axis.jsx

```diff
@@ -507,22 +474,25 @@ export default class VictoryAxis extends React.Component {
       // Do less work by having `VictoryAnimation` tween only values that
       // make sense to tween. In the future, allow customization of animated
       // prop whitelist/blacklist?
-      const animateData = _.omit(this.props, [
-        "orientation", "scale", "tickFormat", "animate",
-        "crossAxis", "standalone"
+      const animateData = _.pick(this.props, [
+        "style", "domain", "range", "tickCount", "tickValues",
+        "labelPadding", "offsetX", "offsetY", "padding", "width", "height"
       ]);
       return (
         <VictoryAnimation {...this.props.animate} data={animateData}>
-          {props => <VictoryAxis {...this.props} {...props} animate={null}/>}
+          {(props) => <VictoryAxis {...this.props} {...props} animate={null}/>}
         </VictoryAnimation>
       );
+    } else {
+      this.getCalculatedValues(this.props);
```

Comment:
I removed `componentWillMount` and `componentWillReceiveProps` and put `getCalculatedValues` here. This aligns with advice I've seen surrounding calculated values: calculate them at the latest possible time. Out of these three lifecycle methods, React is potentially going to call `render` the least.